### ISSUE TITLE
Plans to support the sparse, multi-vector and hybrid portions of BAAI/bge-m3?

### ISSUE THREAD

**Issue by krumeto** [UKPLab/sentence-transformers#2585]
Hey team, are there plans to support the sparse, multi-vector and hybrid portions of `BAAI/bge-m3`?

The individual new components (Colbert-like multi-vector) and sparse ones might be helpful components on their own, and I am curious if support is planned. One can now use `FlagEmbedding`, but would be cool to have the capability in `sentence-transformers` as well.

The current use-case that we have is an internal embedding API for which we support different models. Would be cool not to have to add `FlagEmbedding` as a dependency just to try/test the `BAAI/bge-m3`.

Thank you in advance and thank you for the active development in recent months!

**Comment by tomaarsen** [comment]
Hello!

Sparse and multi-vector (e.g. ColBERT) models are indeed planned, although I suspect that they will use a separate class than `SentenceTransformer` (much like the `CrossEncoder` in `sentence-transformers`) because the outputs are much different, and I want people to be able to rely on the outputs from each class, e.g. `SentenceTransformer` always produces embeddings with `[batch_size, embedding_dimension]` shape.

What I haven't yet decided on is how bge-m3 itself can be integrated. If `SentenceTransformers`, `SparseEmbedding` and `MultiVectorEmbedding` (or whatever they'll be called) are separate classes, then bge-m3 still won't nicely play with it. In truth, I am also not sure whether combined models like bge-m3 will become a norm at all, or whether it makes more sense to use separate models for separate tasks (even if you end up hybrid-ing them together).

In short, this is something that I haven't solved yet. Perhaps bge-m3 will end up staying too niche to warrant an extensive integration, I'm not sure. I'm definitely open to suggestions and discussions here.

- Tom Aarsen

**Comment by krumeto** [comment]
Most of the value would probably be captured just by supporting Sparse and multi-vector. Thank you, please feel free to close the issue.
